### PR TITLE
Merge ScanHub-UI and ScanHub-Backend

### PR DESCRIPTION
**Pull request by schote**
So far the development of scanhub and scanhub-ui was separated in two different repositories.
This was motivated by the fact that potentially other ui's could exist which also use the scanhub backend.
However, we realize that the disadvantages outweigh the advantages of this structure and that it is hard to keep both repositories synchronized. 
We decided to merge the scanhub-ui into the scanhub backend by keeping all of the existing services/structures, i.e. the patient data remains in its separate database. The modular approach still allows to also use other ui's with the scanhub backend.
With the completed merge, we will update the issue list according to the SWRS (update on the documentation pending).

Some of the files must be merged manually:

- [x] Merge docker-compose file
- [x] Merge readme-file
- [x] Merge .gitignore
- [x] Merge/update GitHub workflows

### CODE REVIEW

**Comment by schote** [comment]
Docker-compose command is now working for the merged repository.
docker-container

**Comment by BenWilhelmUniklinikFreiburg** [comment]
There is some issue with the cornerstone viewer. Other than that it looks good and I very much support the idea of the merge of the two repos!

grafik

**Review by BenWilhelmUniklinikFreiburg — APPROVED** [review]
Good, problems with the viewer may be fixed after the merge.

**Comment by schote** [comment]
> There is some issue with the cornerstone viewer. Other than that it looks good and I very much support the idea of the merge of the two repos!

I have just tested it again on my end and it works fine:
Bildschirmfoto 2024-06-19 um 09 49 42

File: frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# FROM node:bullseye-slim AS BUILD_IMAGE
+FROM node:20.5.0-slim AS BUILD_IMAGE
+
+# Install python, required for node-gyp
+# RUN apt-get update || : && apt-get install python -y
```
Comment (chdinh) [inline frontend/Dockerfile:5]: do we still need this commented code

File: frontend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# RUN apt-get --no-cache python3 g++ make
```
Comment (chdinh) [inline frontend/Dockerfile:7]: still required?

File: frontend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+RUN apt-get update || : && apt-get install -y python3 build-essential \
+    && apt-get clean
+
+WORKDIR /app
+COPY package.json yarn.lock ./
+
+# RUN apt update && apk --no-cache --virtual build-dependencies add \
```
Comment (chdinh) [inline frontend/Dockerfile:15]: Still required?

File: frontend/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+#     g++
+
+# Install dependencies, network timeout flag for slow network connection
+# Without frozen lockfile
+# RUN yarn install --network-timeout 100000
+# With frozen lockfile
+RUN yarn install --frozen-lockfile --network-timeout 100000
+
+# Use argument to specify development or production state
+# ARG NODE_ENV=development
```
Comment (chdinh) [inline frontend/Dockerfile:27]: Still required?

File: frontend/src/components/DicomViewer.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                  <CornerstoneViewport
+                    key={index}
+                    imageIds={[`dicomweb:${viewportData[index]}`]}
+                    tools={tools}
+                    setViewportActive={() => {
+                      setActiveViewportIndex(index)
+                    }}
+
+                    // >> Save the following code snippets for later, might be useful
+                    // loadingIndicatorComponent={ CircularProgress }
```
Comment (chdinh) [inline frontend/src/components/DicomViewer.tsx:79]: shall we start a separate snippet galery?

File: frontend/src/components/ExamInstanceItem.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+  const deleteExam = useMutation(async () => {
+    await examApi.examDeleteApiV1ExamExamIdDelete(
+      exam.id, {headers: {Authorization: 'Bearer ' + user?.access_token}}
+    ).then(() => {
+      refetchParentData()
+    })
+  })
+
+  // const updateExam = useMutation(async (data: Exam) => {
```
Comment (chdinh) [inline frontend/src/components/ExamInstanceItem.tsx:48]: debug log

File: frontend/src/generated-client/patient/base.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * ScanHub-UI
```
Comment (chdinh) [inline frontend/src/generated-client/patient/base.ts:4]: header seems to be different compared to our python info headers

File: frontend/src/generated-client/patient/base.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+// Some imports not used depending on template conditions
+// @ts-ignore
+import type { AxiosPromise, AxiosInstance, RawAxiosRequestConfig } from 'axios';
+import globalAxios from 'axios';
+
+export const BASE_PATH = "http://localhost".replace(/\/+$/, "");
+
+/**
+ *
+ * @export
```
Comment (chdinh) [inline frontend/src/generated-client/patient/base.ts:26]: incomplete docu

File: frontend/src/generated-client/patient/base.ts [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+ * @interface RequestArgs
+ */
+export interface RequestArgs {
+    url: string;
+    options: RawAxiosRequestConfig;
+}
+
+/**
```
Comment (chdinh) [inline frontend/src/generated-client/patient/base.ts:45]: incomplete

File: frontend/src/generated-client/patient/base.ts [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    constructor(configuration?: Configuration, protected basePath: string = BASE_PATH, protected axios: AxiosInstance = globalAxios) {
+        if (configuration) {
+            this.configuration = configuration;
+            this.basePath = configuration.basePath ?? basePath;
+        }
+    }
+};
+
+/**
+ *
```
Comment (chdinh) [inline frontend/src/generated-client/patient/base.ts:62]: incomplete

File: frontend/src/generated-client/patient/common.ts [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+ */
+
+
+import type { Configuration } from "./configuration";
+import type { RequestArgs } from "./base";
+import type { AxiosInstance, AxiosResponse } from 'axios';
+import { RequiredError } from "./base";
+
+/**
+ *
```
Comment (chdinh) [inline frontend/src/generated-client/patient/common.ts:22]: docu incomplete throughout the file

File: frontend/src/generated-client/userlogin/base.ts [hunk 12]
```diff
@@ -0,0 +1,86 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * FastAPI
```
Comment (chdinh) [inline frontend/src/generated-client/userlogin/base.ts:4]: check header

File: frontend/src/generated-client/userlogin/common.ts [hunk 13]
```diff
@@ -0,0 +1,150 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * FastAPI
```
Comment (chdinh) [inline frontend/src/generated-client/userlogin/common.ts:4]: check header

File: frontend/src/generated-client/userlogin/common.ts [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+ */
+
+
+import type { Configuration } from "./configuration";
+import type { RequestArgs } from "./base";
+import type { AxiosInstance, AxiosResponse } from 'axios';
+import { RequiredError } from "./base";
+
+/**
+ *
```
Comment (chdinh) [inline frontend/src/generated-client/userlogin/common.ts:22]: docu incomplete throughout the file

**Review by chdinh — CHANGES_REQUESTED** [review]

File: frontend/Dockerfile (hunk 4 above)
Comment (BenWilhelmUniklinikFreiburg) [reply frontend/Dockerfile:19]: is interesting, would keep it

File: frontend/src/components/DicomViewer.tsx (hunk 5 above)
Comment (BenWilhelmUniklinikFreiburg) [reply frontend/src/components/DicomViewer.tsx:79]: no, the snipped is needed in that file, not somewhere else

File: frontend/Dockerfile (hunk 2 above)
Comment (BenWilhelmUniklinikFreiburg) [reply frontend/Dockerfile]: removed

File: frontend/Dockerfile (hunk 3 above)
Comment (BenWilhelmUniklinikFreiburg) [reply frontend/Dockerfile]: removed

File: frontend/src/components/ExamInstanceItem.tsx (hunk 6 above)
Comment (BenWilhelmUniklinikFreiburg) [reply frontend/src/components/ExamInstanceItem.tsx:48]: ?

Comment (BenWilhelmUniklinikFreiburg) [reply frontend/src/components/ExamInstanceItem.tsx:48]: is this really something that should block the merge of the two repos?

File: frontend/src/generated-client/patient/base.ts (hunk 8 above)
Comment (BenWilhelmUniklinikFreiburg) [reply frontend/src/generated-client/patient/base.ts:26]: Yes, but this has nothing to do with the merge. Just regenerated all clients and they were almost completely up to date.

File: frontend/src/generated-client/patient/base.ts (hunk 7 above)
Comment (BenWilhelmUniklinikFreiburg) [reply frontend/src/generated-client/patient/base.ts:4]: suggest handling this with an issue.

**Review by BenWilhelmUniklinikFreiburg — APPROVED** [review]
Looks good!

File: frontend/src/components/ExamInstanceItem.tsx (hunk 6 above)
Comment (schote) [reply frontend/src/components/ExamInstanceItem.tsx:48]: Actually this is not a debug log. The mutation is required to update an exam instance. This feature will be implemented soon.

**Comment by schote** [comment]
Added missing header to scanhub-ui files. Removed some obsolete components and renamed files according to the typescript-react convention.

**Review by chdinh — APPROVED** [review]
Great stuff @schote & @BenWilhelmUniklinikFreiburg 

Looks good to me!

**Pull request merged**